module2/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,69 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+Compressed compressGrayscale(const Image& compress) {
+    Compressed compressedBitmap;
+
+    int repetitionCounter = 0;
+    for (const auto& row : compress) {
+        uint8_t actualColor = row.at(0);
+        auto lineBegin = row.begin();
+        auto lineEnd = row.end();

```

Comment:
You don't need it here. You can move this to first loop

Reply:
@nauka-programowania-MA can you explain? :) Because these code lines in the first loop :)